lib/pbench/test/unit/server/test_requests.py

```diff
@@ -90,13 +90,17 @@ def test_bad_extension_upload(client, bad_extension):
 
     @staticmethod
     def test_upload(client, pytestconfig):
-        filename = "./lib/pbench/test/unit/server/fixtures/upload/log.tar.xz"
+        datafile = "./lib/pbench/test/unit/server/fixtures/upload/log.tar.xz"
+        with open(datafile, "rb") as f:
+            data = f.read()
 
-        with open(f"{filename}.md5") as md5sum_check:
+        with open(f"{datafile}.md5") as md5sum_check:
             md5sum = md5sum_check.read()
 
+        filename = "log.tar.xz"
         response = client.post(
             f"{client.config['REST_URI']}/upload",
+            data=data,
```

Comment:
Can `data` be the file object?  I seem to recall that you can have the `post` method read the file.

Reply:
yes good point I think I can do `data = open(datafile, "rb")`